src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCacheableRequest.java

```diff
@@ -132,109 +115,44 @@
-                                                    connectReply,
-                                                    connectReplyId,
-                                                    connectRouteId,
-                                                    ifNoneMatch,
-                                                    this::scheduleRequest);
-            newStream = cacheableResponse::onResponseMessage;
+            newStream = new HttpCacheProxyNotModifiedResponse(factory,
+                                                             requestGroup.getRequestHash(),
+                                                             prefer,
+                                                             accept,
+                                                             acceptRouteId,
+                                                             acceptReplyId,
+                                                             connect,
+                                                             connectReplyId,
+                                                             connectRouteId)::onResponseMessage;
         }
         else
         {
-            final HttpCacheProxyNonCacheableResponse nonCacheableResponse =
-                    new HttpCacheProxyNonCacheableResponse(factory,
-                                                           connectReply,
-                                                           connectRouteId,
-                                                           connectReplyId,
-                                                           acceptReply,
-                                                           acceptRouteId,
-                                                           acceptReplyId);
-            factory.router.setThrottle(acceptReplyId, nonCacheableResponse::onResponseMessage);
-            newStream = nonCacheableResponse::onResponseMessage;
-            requestGroup.onNonCacheableResponse(acceptReplyId);
-            cleanupRequestIfNecessary();
-            resetRequestTimeoutIfNecessary();
+            newStream = new HttpCacheProxyNonCacheableResponse(factory,
+                                                               connect,
+                                                               connectRouteId,
+                                                               connectReplyId,
+                                                               accept,
+                                                               acceptRouteId,
+                                                               acceptReplyId)::onResponseMessage;
         }
 
-        return newStream;
-    }
+        connect = factory.router.supplyReceiver(connectReplyId);
+        factory.router.setThrottle(acceptReplyId, newStream);
+        cleanupRequestSlotIfNecessary();
+        resetRequestTimeoutIfNecessary();
+        requestGroup.dequeue(ifNoneMatch, acceptReplyId);
```

Comment:
Please add `assert newStream != null` above this block of code.

Reply:
Done.